src/ORM/Association.php

```diff
@@ -359,14 +359,14 @@ public function bindingKey($key = null)
      */
     public function foreignKey($key = null)
     {
-        if ($key !== null) {
+        if (func_num_args() === 1) {
```

Comment:
It would be safer to check for `!== 0` as that is more equal to the strict null check. What if someone calls it with two arguments?